src/backends/onnxruntime.c

```diff
@@ -464,33 +565,30 @@ int RAI_ModelRunORT(RAI_ModelRunCtx *mctx, RAI_Error *error)
     status = ort->Run(session, run_options, input_names, (const OrtValue *const *)inputs,
                      n_input_nodes, output_names, n_output_nodes, outputs);
 
-    if (status)
-    {
+    if (status) {
       goto error;
     }
 
-    for (size_t i = 0; i < n_output_nodes; i++)
-    {
-      RAI_Tensor *output_tensor = RAI_TensorCreateFromOrtValue(outputs[i], error);
-      if (error->code != RAI_OK)
-      {
-        ort->ReleaseStatus(status);
-        return 1;
-      }
-      if (output_tensor)
-      {
-        mctx->outputs[i].tensor = RAI_TensorGetShallowCopy(output_tensor);
-        RAI_TensorFree(output_tensor);
-      }
-      else
-      {
-        printf("ERR: non-tensor output from ONNX models, ignoring (currently unsupported).\n");
+    for (size_t i = 0; i < n_output_nodes; i++) {
+      for (size_t b=0; b<nbatches; b++) {
+        RAI_Tensor* output_tensor = RAI_TensorCreateFromOrtValue(outputs[i], batch_offsets[b], batch_sizes[b], error);
+        if (error->code != RAI_OK) {
+          ort->ReleaseStatus(status);
+          return 1;
+        }
+        if (output_tensor) {
+          mctx->batches[b].outputs[i].tensor = RAI_TensorGetShallowCopy(output_tensor);
+          RAI_TensorFree(output_tensor);
+        }
+        else {
+          printf("ERR: non-tensor output from ONNX models, ignoring (currently unsupported).\n");
```

Comment:
should we log it?

Reply:
This is a good point, that might lead to a minor refactoring.

In order to Log, we need `ctx`, which we don't have (and I'd like to avoid). The way we usually do it for errors is populating the `RAI_Error` data structure and check it on the caller's end. We could do the same here, but we need to handle the fact that `RAI_Error` could contain a warning (e.g. recognize it based on the status and just log), which is something we need to handle on the caller.

We can do this as a general mechanism, does it sound good?

Reply:
As discussed let's create an issue and resolve this in a more broader way in the future